caddyconfig/httpcaddyfile/directives.go

```diff
@@ -250,6 +251,63 @@ func (h Helper) NewBindAddresses(addrs []string) []ConfigValue {
 	return []ConfigValue{{Class: "bind", Value: addrs}}
 }
 
+// ParseSegmentAsSubroute parses the segment such that its subdirectives
+// are themselves treated as directives, from which a subroute is built
+// and returned.
+func ParseSegmentAsSubroute(h Helper) (caddyhttp.MiddlewareHandler, error) {
```

Comment:
The only change here is `parseSegmentAsSubroute` -> `ParseSegmentAsSubroute`, but I moved the function up in the file because I think the convention is to keep exported functions together?

Reply:
Sure -- doesn't matter too much. The go std lib puts functions all over the place, doesn't really matter if you know how to use your editor in the slightest.